core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/jdbc/JDBCUserStoreManager.java

```diff
@@ -324,14 +337,61 @@ public JDBCUserStoreManager(RealmConfiguration realmConfig, Map<String, Object>
         }
 
         initUserRolesCache();
-
+        if (hashProvider == null) {
+            hashProvider = initializeHashProvider(realmConfig.getUserStoreProperties());
+        }
         if (log.isDebugEnabled()) {
             log.debug("Ended " + System.currentTimeMillis());
         }
         /* Initialize user roles cache as implemented in AbstractUserStoreManager */
 
     }
 
+    /**
+     * Initialize the HashProvider according to the given user store properties.
+     *
+     * @param userStorePropertiesMap User store properties.
+     * @return Initialized HashProvider.
+     */
+    private HashProvider initializeHashProvider(Map<String, String> userStorePropertiesMap) {
+
+        String digestFunction = userStorePropertiesMap.get(JDBCRealmConstants.DIGEST_FUNCTION);
+        HashProviderFactory hashProviderFactory =
+                UserStoreMgtDataHolder.getInstance().getHashProviderFactory(digestFunction);
+        if (hashProviderFactory == null) {
+            hashProvider = null;
+        } else {
+            Set<String> metaProperties = hashProviderFactory.getHashProviderMetaProperties();
+            if (metaProperties.isEmpty()) {
+                hashProvider = hashProviderFactory.getHashProvider();
+            } else {
+                Map<String, Object> hashProviderPropertiesMap =
+                        getHashProviderInitConfigs(userStorePropertiesMap);
+                hashProvider = hashProviderFactory.getHashProvider(hashProviderPropertiesMap);
+            }
+        }
+        return hashProvider;
+    }
+
+    /**
+     * Get map of user store properties related to the HashProviderFactory.
+     *
+     * @param userStorePropertiesMap The map which contains the user store properties.
+     * @return The map of user store properties related to the HashProviderFactory.
+     */
+    private Map<String, Object> getHashProviderInitConfigs(Map<String, String> userStorePropertiesMap) {
+
+        String hashingAlgorithmProperties = userStorePropertiesMap.get(JDBCRealmConstants.HASHING_ALGORITHM_PROPERTIES);
```

Comment:
If this is blank we can stop from here, right?

Reply:
it cannot be blank. It is a required property.

Reply:
resolved